components/google-cloud/tests/aiplatform/test_remote_runner.py

```diff
@@ -13,10 +13,272 @@
+            json.dumps(expected_output)
+        )
+
+    @mock.patch('builtins.open', new_callable=mock.mock_open())
+    @mock.patch.object(os, "makedirs", autospec=True)
+    def test_write_to_artifact_with_no_artifacts_writes_emtpy_dict(
+        self, mock_makedirs, mock_open
+    ):
+        remote_runner.write_to_artifact({
+            "outputs": {
+                "artifacts": {},
+                "outputFile": "/gcs/test_path/executor_output.json"
+            }
+        }, "test_resource_uri")
+
+        mock_open.return_value.__enter__().write.assert_called_once_with(
+            json.dumps({})
+        )
+
+    def test_resolve_input_args_non_resouce_noun_removes_gcs_prefix(self):
+        type_to_resolve = aiplatform.base.AiPlatformResourceNoun
+        value = '/gcs/test_resouce_name'
+        expected_result = 'test_resouce_name'
+
+        result = remote_runner.resolve_input_args(value, type_to_resolve)
+        self.assertEqual(result, expected_result)
+
+    def test_resolve_input_args_resouce_noun_not_changed(self):
+        type_to_resolve = aiplatform.base.AiPlatformResourceNoun
+        value = 'test_resouce_name'
+        expected_result = 'test_resouce_name'
+
+        result = remote_runner.resolve_input_args(value, type_to_resolve)
+        self.assertEqual(result, expected_result)
+
+    def test_resolve_input_args_not_type_to_resolve_not_changed(self):
+        type_to_resolve = object
+        value = '/gcs/test_resouce_name'
+        expected_result = '/gcs/test_resouce_name'
+
+        result = remote_runner.resolve_input_args(value, type_to_resolve)
+        self.assertEqual(result, expected_result)
+
+    def test_resolve_init_args_key_does_not_end_with_name_not_changed(self):
+        key = 'resouce'
+        value = '/gcs/test_resouce_name'
+        expected_result = '/gcs/test_resouce_name'
+
+        result = remote_runner.resolve_init_args(key, value)
+        self.assertEqual(result, expected_result)
+
+    def test_resolve_init_args_key_ends_with_name_removes_gcs_prefix(self):
+        key = 'resouce_name'
+        value = '/gcs/test_resouce_name'
+        expected_result = 'test_resouce_name'
+
+        result = remote_runner.resolve_init_args(key, value)
+        self.assertEqual(result, expected_result)
+
+    @mock.patch.object(
+        utils,
+        "is_mb_sdk_resource_noun_type",
+        autospec=True,
+        return_value=False
+    )
+    def test_make_output_with_not_resouce_name_returns_serialized_list_json(
+        self, mock_utils
+    ):
+        output_object = ["a", 2]
+        expected_result = '["a", 2]'
+        result = remote_runner.make_output(output_object)
+        self.assertEqual(result, expected_result)
+
+    @mock.patch.object(
+        utils, "is_mb_sdk_resource_noun_type", autospec=True, return_value=True
+    )
+    def test_make_output_with_resouce_name_returns_resouce_name_value(
+        self, mock_utils
+    ):
+        output_object = mock.Mock()
```

Comment:
If you wanted to test this without mocking:

```python
output_object = aiplatform.Model._empty_constructor(project='test-project')
output_object._gca_resource = aiplatform.gapic.Model(name='resource_name')
```

Reply:
Done! nice I like using the actual method in this case over mock object, thank you.